test_server.py

```diff
@@ -71,91 +85,185 @@ def parse_response(response):
         * Response include valid date
         * Response includes valid status code
 
-    if these validations are met, then return status code from response"""
+    if these validations are met, then return status code from response
+    as well as the body as (status_code, body)"""
     response = response.strip(CRLF).strip()
     lines = response.split(CRLF)
     initial_line = lines[0]
-    # Get method from initial line, and strip any leading white-space
-    # or CRLF chars. Also format to uppercase for consistent handling.
     protocol = initial_line.split()[0].strip().upper()
     response_code = initial_line.split()[1].strip()
-    # response_code = initial_line.split()[2].strip()
-
-    #  Get headers by splitting response by CRLF and dropping the first line.
-    headers = [line.split()[0].strip() for line in lines]
-    #  Grabbing each header from above, removing trailing colon and converting to uppercase
-    headers = [header.rstrip(':').upper() for header in headers]
-    #  Converting headers to set for ease of membership testing
+    headers = [line for line in lines if b':' in line]
+    headers = [line.split(b':')[0].strip().upper() for line in lines]
     headers = set(headers)
 
     try:
         int(response_code)
     except ValueError:
-        to_return = False
+        to_return = False, None
     else:
         if b'DATE' not in headers:
-            to_return = False
+            to_return = False, None
         elif b'HTTP/1.1' not in protocol:
-            to_return = False
+            to_return = False, None
         else:
             #  HTTP response passes all prior checks, pass response code back
-            to_return = response_code
+            to_return = response_code, lines[-1]
     return to_return
 
 
+###################################################
+# Tests of Functions
+###################################################
 def test_start_server(server_process):
     """Dummy function to start server"""
     pass
 
 
 def test_parse_good_request():
-    assert server.parse_request(REQ_GOOD) == "wwww.host.com/stuff"
+    assert server.parse_request(REQ_GOOD) == b"http://www.host.com/images"
 
 
 def test_parse_bad_method_request():
     with pytest.raises(IndexError):
-      server.parse_request(REQ_BAD_METHOD)
+        server.parse_request(REQ_BAD_METHOD)
 
 
 def test_parse_bad_protocol_request():
     with pytest.raises(NotImplementedError):
-      server.parse_request(REQ_BAD_PROTOCOL)
+        server.parse_request(REQ_BAD_PROTOCOL)
 
 
 def test_parse_bad_host_request():
     with pytest.raises(ValueError):
-      server.parse_request(REQ_BAD_HOST)
+        server.parse_request(REQ_BAD_HOST)
 
 
 def test_response_ok():
-    foo_uri = 'www.host.com/stuff'
+    foo_uri = b'http://www.host.com'
```

Comment:
Proper URIs should really only be the Path portion of a URL, not the protocol or domain.